lib/rules/modifier-name-case.js

```diff
@@ -12,15 +13,37 @@ export default class ModifierNameCase extends Rule {
       ElementModifierStatement(node) {
         let modifierName = node.path.original;
 
-        if (modifierName === dasherize(modifierName)) {
+        if (typeof modifierName === 'string') {
+          this._validateModifierName(modifierName, node.path);
+        }
+      },
+
+      SubExpression(node) {
+        if (isModifierHelper(node) === false) {
           return;
         }
 
-        this.log({
-          message: generateErrorMessage(modifierName),
-          node: node.path,
-        });
+        let nameParam = node.params[0];
+
+        if (nameParam) {
+          this._validateModifierName(nameParam.original, nameParam);
+        }
       },
     };
   }
+
+  _validateModifierName(modifierName, node) {
```

Comment:
Maybe just pass in a single `node` parameter and then create the `modifierName` inside the function?